backend/internetarchive/internetarchive.go

```diff
@@ -0,0 +1,1086 @@
+// Package internetarchive provides an interface to Internet Archive's Item
+// via their native API than using S3-compatible endpoints.
+package internetarchive
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ncw/swift/v2"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/fserrors"
+	"github.com/rclone/rclone/fs/fshttp"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/lib/bucket"
+	"github.com/rclone/rclone/lib/encoder"
+	"github.com/rclone/rclone/lib/pacer"
+	"github.com/rclone/rclone/lib/rest"
+)
+
+// Register with Fs
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "internetarchive",
+		Description: "Internet Archive",
+		NewFs:       NewFs,
+		Options: []fs.Option{{
+			Name: "access_key_id",
+			Help: "IAS3 Access Key.\n\nLeave blank for anonymous access.\nYou can find one here: https://archive.org/account/s3.php",
+		}, {
+			Name: "secret_access_key",
+			Help: "IAS3 Secret Key (password).\n\nLeave blank for anonymous access.",
+		}, {
+			// their official client (https://github.com/jjjake/internetarchive) hardcodes following the two
+			Name:     "endpoint",
+			Help:     "IAS3 Endpoint.\n\nLeave blank for default value.",
+			Default:  "https://s3.us.archive.org",
+			Advanced: true,
+		}, {
+			Name:     "front_endpoint",
+			Help:     "Host of InternetArchive Frontend.\n\nLeave blank for default value.",
+			Default:  "https://archive.org",
+			Advanced: true,
+		}, {
+			Name: "disable_checksum",
+			Help: `Don't ask the server to test against MD5 checksum calculated by rclone.
+Normally rclone will calculate the MD5 checksum of the input before
+uploading it so it can ask the server to check the object against checksum.
+This is great for data integrity checking but can cause long delays for
+large files to start uploading.`,
+			Default:  true,
+			Advanced: true,
+		}, {
+			Name: "wait_archive",
+			Help: `Wait for the archive and book_op tasks to finish. 
+This tries to wait at its best, but not complete.
+Only enable if you need to be guaranteed to be reflected after write operations.`,
+			Default:  false,
+			Advanced: true,
+		}, {
+			Name:     config.ConfigEncoding,
+			Help:     config.ConfigEncodingHelp,
+			Advanced: true,
+			Default: encoder.EncodeZero |
+				encoder.EncodeSlash |
+				encoder.EncodeLtGt |
+				encoder.EncodeCrLf |
+				encoder.EncodeDel |
+				encoder.EncodeCtl |
+				encoder.EncodeLeftCrLfHtVt |
```

Comment:
I wouldn't have thought you would need EncodeLeftCrLfHtVT and EncodeRIghtCrLfHtVt as well as EncodeCtrl since EncodeCtrl covers CR, LF, HT and VT?

Reply:
Ah yes, `EncodeCtl` is enough